app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala

```diff
@@ -30,40 +28,41 @@ import play.api.http.Status.BAD_GATEWAY
 
 import scala.concurrent.{ExecutionContext, Future}
 
-//TODO[Testing] this class needs testing
-class CitizenDetailsConnector @Inject() (
+class CitizenDetailsConnector @Inject()(
   http: HttpClient,
   metricsService: MetricsService,
   appConfig: ApplicationConfig
 )(implicit executionContext: ExecutionContext) {
 
   val serviceUrl: String = appConfig.citizenDetailsServiceUrl
 
-  def connectToGetPersonDetails(nino: Nino)(
-    implicit hc: HeaderCarrier): Future[Either[UpstreamErrorResponse, CitizenDetailsResponse]] = {
+  def connectToGetPersonDetails(
+    nino: Nino
+  )(
+    implicit hc: HeaderCarrier
+  ): Future[Either[UpstreamErrorResponse, CitizenDetailsResponse]] = {
 
     val context = metricsService.citizenDetailsTimer.time()
 
     http
       .GET[Either[UpstreamErrorResponse, HttpResponse]](
         s"$serviceUrl/citizen-details/$nino/designatory-details"
-      )
-      .transform { result =>
+      ).transform {
+      result =>
         context.stop()
         result
-      }
-      .map {
-        case Right(response) => Right(response.json.as[CitizenDetailsResponse])
-        case Left(error) =>
-          metricsService.citizenDetailsFailedCounter.inc()
-          Left(error)
-      }
-      .recover {
-        case error: HttpException =>
-          Left(UpstreamErrorResponse(error.message, BAD_GATEWAY))
-        case error =>
-          metricsService.citizenDetailsFailedCounter.inc()
-          throw error
-      }
+    } map {
+      case Right(response) =>
+        Right(response.json.as[CitizenDetailsResponse])
+      case Left(error) =>
+        metricsService.citizenDetailsFailedCounter.inc()
+        Left(error)
+    } recover {
+      case error: HttpException =>
+        Left(UpstreamErrorResponse(error.message, BAD_GATEWAY))
```

Comment:
citizenDetailsFailedCounter.inc() is missing

Reply:
I haven't changed this, just refactored to make more readable

Reply:
Grafana dashboard does not display this metric anyway...